pkg/kubelet/dockertools/manager.go

```diff
@@ -1261,33 +1271,46 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod) err
 				return
 			}
 
+			killContainerResult := &SyncResult{Action: KillContainer, Target: container.Name}
```

Comment:
@Random-Liu It's not your fault, but we need to remove `wg.Add(1)` in to loop and use `wg.Add(len(runningPod.Containers)` instead, because otherwise it will make race detector angry for go 1.5. See https://groups.google.com/forum/#!topic/golang-nuts/q7gmFZEF618